bioconda_utils/docker_utils.py

```diff
@@ -0,0 +1,325 @@
+            Tag to be used for the custom-build docker container. Mostly for
+            debugging purposes when you need to inspect the container.
+
+        container_recipe : str
+            Directory to which the host's recipe will be exported. Will be
+            read-only.
+
+        container_staging : str
+            Directory to which the host's conda-bld dir will be mounted so that
+            the container can use previously-built packages as depdendencies.
+            Upon successful building container-built packages will be copied
+            over. Mounted as read-write.
+
+        image : str
+            Base image from which the new custom image will be built (used on
+            the `FROM:` line of the Dockerfile)
+
+        base_url : str
+            Passed to docker.Client()
+
+        requirements : None or str
+            Path to a "requirements.txt" file which will be installed with
+            conda in a newly-created container. If None, then use the default
+            installed with bioconda_utils.
+
+        verbose : bool
+        """
+        self.image = image
+        self.tag = tag
+        self.verbose = verbose
+        self.requirements = requirements
+        uid = os.getuid()
+        usr = pwd.getpwuid(uid)
+        self.user_info = dict(
+            uid=uid,
+            gid=usr.pw_gid,
+            groupname=grp.getgrgid(usr.pw_gid).gr_name,
+            username=usr.pw_name)
+        self.container_recipe = container_recipe
+        self.container_staging = container_staging
+
+        # Don't import until here to allow module import if docker not
+        # installed.
+        try:
+            from docker import Client as DockerClient
+        except ImportError:
+            raise ImportError("docker-py module must be installed")
+        self.docker = DockerClient(base_url=base_url)
+        self.host_conda_bld = get_host_conda_bld()
+        self._build_container()
+
+    def _build_container(self):
+        """
+        Builds a new image with requirements installed.
+        """
+
+        # Create a temporary build directory since we'll be copying the
+        # requirements file over
+        build_dir = tempfile.mkdtemp()
+
+        logger.debug('Building image "%s" from %s', self.tag, build_dir)
+        with open(os.path.join(build_dir, 'requirements.txt'), 'w') as fout:
+            if self.requirements:
+                fout.write(open(self.requirements).read())
+            else:
+                fout.write(open(pkg_resources.resource_filename(
+                    'bioconda_utils', 'bioconda_utils-requirements.txt')).read())
+
+        with open(os.path.join(build_dir, "Dockerfile"), 'w') as fout:
+            fout.write(dedent(
+                """\
+                FROM {self.image}
+                COPY requirements.txt /tmp/requirements.txt
+                RUN /opt/conda/bin/conda install --file /tmp/requirements.txt
+                """.format(**locals())))
```

Comment:
Are you building a new container in order to avoid redundancy? Otherwise, wouldn't it be possible to just prepend those commands to the actual conda build command?